vanillaSelectBox.js

```diff
@@ -328,6 +331,23 @@ function vanillaSelectBox(domSelector, options) {
                         }
                     });
                 }
+                if (self.optionNodesByGroup) {
+                    self.optionNodesByGroup.forEach(function (optGroup) {
+                        let hideOptGroup = true;
```

Comment:
i think you can just ask for ``let hideOptGroup = optGroup.querySelectorAll('.hidden-search').length === optGroup.querySelectorAll('li').length``, that means that all children are hidden, we don't need a loop or even store children options